src/ast/ast_validations.c

```diff
@@ -1196,7 +1196,11 @@ static void _AST_GetDefinedIdentifiers(const cypher_astnode_t *node, rax *identi
 		 * UNWIND [ref_1, ref_2] AS defined RETURN defined */
 		const cypher_astnode_t *unwind_alias_node = cypher_ast_unwind_get_alias(node);
 		const char *unwind_alias = cypher_ast_identifier_get_name(unwind_alias_node);
-		raxInsert(identifiers, (unsigned char *)unwind_alias, strlen(unwind_alias), NULL, NULL);
+		int rc = raxTryInsert(identifiers, (unsigned char *)unwind_alias,
+							  strlen(unwind_alias), NULL, NULL);
+		if(rc == 0) {
+			ErrorCtx_SetError("Variable %s already declared", unwind_alias);
+		}
```

Comment:
This check is misplaced
 `_AST_GetDefinedIdentifiers` shouldn't deal with error reporting.
 
 We need a single function which verifies variable overloading / redefinition for all relevant clauses for a given scope.

Clauses: `MATCH, CREATE, MERGE, UNWIND and CALL`
\* I might be missing a clause which can introduce variables.